internal/config/cgmgr/cgroupfs.go

```diff
@@ -0,0 +1,83 @@
+// +build linux
+
+package cgmgr
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/libpod/pkg/cgroups"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type CgroupfsManager struct{}
+
+const (
+	defaultCgroupfsParent = "/crio"
+)
+
+// Name returns the name of the cgroup manager (cgroupfs)
+func (*CgroupfsManager) Name() string {
+	return cgroupfsCgroupManager
+}
+
+// IsSystemd returns that this is not a systemd cgroup manager
+func (*CgroupfsManager) IsSystemd() bool {
+	return false
+}
+
+// ContainerCgroupPath takes arguments sandbox parent cgroup and container ID and returns
+// the cgroup path for that containerID. If parentCgroup is empty, it
+// uses the default parent /crio
+func (*CgroupfsManager) ContainerCgroupPath(sbParent, containerID string) string {
+	parent := defaultCgroupfsParent
+	if sbParent != "" {
+		parent = sbParent
+	}
+	return filepath.Join(parent, scopePrefix+"-"+containerID)
+}
+
+// SandboxCgroupPath takes the sandbox parent, and sandbox ID. It
+// returns the cgroup parent, cgroup path, and error.
+func (*CgroupfsManager) SandboxCgroupPath(sbParent, sbID string) (cgParent, cgPath string, err error) {
+	if strings.HasSuffix(path.Base(sbParent), ".slice") {
+		return "", "", fmt.Errorf("cri-o configured with cgroupfs cgroup manager, but received systemd slice as parent: %s", sbParent)
```

Comment:
One thing we have discussed in the past is the removal of this logic entirely. If the kubelet sends us a slice we pick the systemd cgroup manager and if we doesn't then we pick the cgroupfs manager.

Reply:
I've always wondered why we didn't do that. would make for a good follow up :smiley: